bf: ZENKO-1736 count items scan

philipyoo opened the pull request:
Script is mainly to be used by a kubernetes cron job to perform the aggregate query scan 

Also added a change to dockerfile, necessary for building

philipyoo committed:
bf: ZENKO-1736 add scan mongo count items script

This is mainly to be used by a kubernetes cron job to
perform the aggregate query scan and update to mongo
infostore collection.

bert-e commented:
# Hello philipyoo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/s3utils/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @rahulreddy

* @ssalaues

* @rachedbenmustapha

* @vrancurel

bennettbuchanan commented on countItems.js:
```diff
@@ -0,0 +1,33 @@
+const mongoclient = new MongoClientInterface(config);
+mongoclient.setup(err => {
+    if (err) {
+        return log.error('error connecting to mongodb', err);
```
`err` => `{ error: err }`

alexanderchan-scality commented on countItems.js:
```diff
@@ -0,0 +1,33 @@
+        }
+        log.info('mongo scan item count has completed');
+        return process.exit(0);
+    });
```
remarks: not sure how mongodb handles dead connections, so manually closing the connection maybe a good idea

philipyoo committed:
bf: ZENKO-1736 update dockerfile

jessie/mongodb-org -> stretch/mongodb-org
to support the change to node:8-slim and necessary to build
dockerfile

philipyoo commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/7.4`
* `development/8.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/7.4`
* `development/8.0`



Please check the status of the associated issue ZENKO-1736.

Goodbye philipyoo.
